cmake/GoogleCloudCppDoxygen.cmake

```diff
@@ -97,49 +113,69 @@ elseif (GOOGLE_CLOUD_CPP_GENERATE_DOXYGEN)
-    endif ()
+    google_cloud_cpp_doxygen_deploy_version(VERSION)
     list(
         APPEND
         DOXYGEN_ALIASES
-        "googleapis_dev_link{2}=\"https://googleapis.dev/cpp/google-cloud-\\1/${GOOGLE_CLOUD_CPP_DOXYGEN_VERSION}/\\2\""
+        "googleapis_dev_link{2}=\"https://googleapis.dev/cpp/google-cloud-\\1/${VERSION}/\\2\""
     )
-    set(GOOGLE_CLOUD_CPP_DOXYGEN_INPUTS "${CMAKE_CURRENT_SOURCE_DIR}")
-    if ("cloud" STREQUAL "${GOOGLE_CLOUD_CPP_SUBPROJECT}")
-        # We cannot recurse from the google/cloud directory, because it will
-        # traverse all of the libraries. So we turn off recursion and manually
-        # provide the subdirectories to be traversed.
-        set(DOXYGEN_RECURSIVE NO)
-        list(APPEND GOOGLE_CLOUD_CPP_DOXYGEN_INPUTS
-             "${CMAKE_CURRENT_SOURCE_DIR}/mocks"
-             "${CMAKE_CURRENT_SOURCE_DIR}/doc")
-    else ()
+    set(GOOGLE_CLOUD_CPP_DOXYGEN_INPUTS ${_opt_INPUTS})
+    if (opt_RECURSIVE)
         set(DOXYGEN_RECURSIVE YES)
-        set(DOXYGEN_TAGFILES
-            "${GOOGLE_CLOUD_CPP_COMMON_TAG}=https://googleapis.dev/cpp/google-cloud-common/${GOOGLE_CLOUD_CPP_DOXYGEN_VERSION}/"
-        )
+    else ()
+        set(DOXYGEN_RECURSIVE NO)
     endif ()
+    set(DOXYGEN_TAGFILES ${opt_TAGFILES})
     doxygen_add_docs(
-        ${GOOGLE_CLOUD_CPP_SUBPROJECT}-docs "${GOOGLE_CLOUD_CPP_DOXYGEN_INPUTS}"
+        ${library}-docs "${opt_INPUTS}"
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        COMMENT "Generate ${GOOGLE_CLOUD_CPP_SUBPROJECT} HTML documentation")
-    add_dependencies(doxygen-docs ${GOOGLE_CLOUD_CPP_SUBPROJECT}-docs)
+        COMMENT "Generate ${library} HTML documentation")
+    add_dependencies(doxygen-docs ${library}-docs)
 
     # Extra dependencies needed by this subproject's docs target.
-    if (GOOGLE_CLOUD_CPP_DOXYGEN_DEPS)
-        add_dependencies(${GOOGLE_CLOUD_CPP_SUBPROJECT}-docs
-                         ${GOOGLE_CLOUD_CPP_DOXYGEN_DEPS})
-    endif ()
-    if (NOT ("cloud" STREQUAL "${GOOGLE_CLOUD_CPP_SUBPROJECT}"))
-        add_dependencies(${GOOGLE_CLOUD_CPP_SUBPROJECT}-docs "cloud-docs")
+    if (opt_DEPENDS)
+        add_dependencies(${library}-docs ${opt_DEPENDS})
     endif ()
+endfunction ()
+
+function (google_cloud_cpp_doxygen_targets library)
+    google_cloud_cpp_doxygen_deploy_version(VERSION)
+    set(GOOGLE_CLOUD_CPP_COMMON_TAG
+        "${PROJECT_BINARY_DIR}/google/cloud/cloud.tag")
+    cmake_parse_arguments(opt "" "" "INPUTS;TAGFILES;DEPS" ${ARGN})
+    google_cloud_cpp_doxygen_targets_impl(
+        "${library}"
+        RECURSIVE
+        INPUTS
+        "${CMAKE_CURRENT_SOURCE_DIR}"
+        DEPENDS
+        "cloud-docs"
+        ${GOOGLE_CLOUD_CPP_DOXYGEN_DEPS}
+        TAGFILES
+        "${GOOGLE_CLOUD_CPP_COMMON_TAG}=https://googleapis.dev/cpp/google-cloud-common/${VERSION}/"
+    )
+endfunction ()
+
+# TODO(#10519) - have each subproject call this directly Find out the name of
```

Comment:
s/Find/to find/?

Reply:
Argh... cmake-format joined two unrelated comments.  I am going to beat it with the `# ~~~` hammer.  That will teach it.